File: backend/schemas/resume.py
from pydantic import BaseModel


class ResumeExperience(BaseModel):
    title: str
    company: str
    start_date: str
    end_date: str | None = None
    description: str


class ResumeEducation(BaseModel):
    degree: str
    institution: str
    graduation_year: str


class ResumeResponse(BaseModel):
    id: str
    user_id: str
    filename: str
    file_path: str
    raw_text: str
    parsed_skills: list[str]
    parsed_experience: list[ResumeExperience]
    parsed_education: list[ResumeEducation]
    parsed_achievements: list[str]
    uploaded_at: str
    updated_at: str

    class Config:
        from_attributes = True

File: packages/shared/src/types/resume.ts
export interface ResumeExperience {
  title: string;
  company: string;
  start_date: string;
  end_date?: string;
  description: string;
}

export interface ResumeEducation {
  degree: string;
  institution: string;
  graduation_year: string;
}

export interface Resume {
  id: string;
  user_id: string;
  filename: string;
  file_path: string;
  raw_text: string;
  parsed_skills: string[];
  parsed_experience: ResumeExperience[];
  parsed_education: ResumeEducation[];
  parsed_achievements: string[];
  uploaded_at: string;
  updated_at: string;
}
